### PR TITLE
feat(publish): support git "describeTag" configuration in version/publish commands

### LINKED ISSUE #507: "--include-merged-tags" not working in "lerna publish" and "lerna version"
### Describe the bug

when use "lerna changed",it is found 1 package ready to publish, but when use  "lerna publish" or  "erna version", it is assuming all packages changed

### Expectation

 In the above questions，found 1 package ready to publish is right, so expect the "lerna publish" and "lerna version" behavior to be consistent with the "lerna changed" behavior

### Reproduction

Git log need contain "Merge branch xxx".

The following steps may be required for problem recurrence:
1. Create a new branch
2. publish a package,
3. merge the branch into the master,
4. create another branch,
5. and change another package for publishing

### Lerna config and logs

<details><summary>lerna.json</summary><p>

```json
<!-- Please paste your `lerna.json` here -->
```
</p></details>

<details><summary>lerna-debug.log</summary><p>

```txt
<!-- If you have a `lerna-debug.log` available, please paste it here -->
<!-- Otherwise, feel free to delete this <details> block -->
```
</p></details>

### Environment Info

  System:
    OS: macOS 11.3.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 126.13 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
 
 Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.13.0/bin/yarn
    npm: 8.1.0 - ~/.nvm/versions/node/v16.13.0/bin/npm
  
Browsers:
    Chrome: 110.0.5481.177
    Firefox: 78.9.0
    Safari: 14.1

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/lerna-lite/lerna-lite/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://github.com/lerna-lite/lerna-lite#readme).
- [X] Check that there isn't [already an issue](https://github.com/lerna-lite/lerna-lite/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/lerna-lite/lerna-lite/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by xueran**
## Description
Add `descripteTag` in the `lerna.json` configuration to allow users to customize the tag pattern. If `descripteTag` is not configured, the tag pattern defaults to`*@*`.

However, it is not supported in `lerna publish from-git`, only in `version`, `publish -- canary`

## Motivation and Context

Add feature to fix #507 

## Types of changes

- [ ] Chore (change that has absolutely no effect on users)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by ghiscoding** [comment]
Thanks for the contribution, it seems the linting is failing, could you run the `prettier:write` pnpm script to fix the linting?

https://github.com/lerna-lite/lerna-lite/blob/f2c91cee5a06236b8fc229eb1aac7591d91398ef/package.json#L33-L34

File: packages/core/src/models/interfaces.ts [hunk 1]
```diff
@@ -187,6 +187,9 @@ export interface ProjectConfig extends LernaConfig, QueryGraphConfig {
 
   /** callback to execute when Promise resolved */
   onResolved?: (result: any) => void;
+
+  /** custom tag pattern, default is "*@*" */
```
Comment (ghiscoding) [inline packages/core/src/models/interfaces.ts:191]: the default of `*@*` is only for independent mode, it's empty for version mode.

File: packages/core/src/utils/collect-updates/collect-updates.ts [hunk 2]
```diff
@@ -44,7 +45,10 @@ export function collectUpdates(
       : new Map(filteredPackages.map(({ name }) => [name, packageGraph.get(name)]));
 
   let committish = commandOptions.since;
-  const tagPattern = isIndependent ? '*@*' : '';
+  let tagPattern = '';
+  if (isIndependent) {
+    tagPattern = describeTag ? describeTag : '*@*';
```
Comment (ghiscoding) [inline packages/core/src/utils/collect-updates/collect-updates.ts:50]: so from what I see, this describe will only work for the `independent` mode and not the version mode, I wonder if it wouldn't be better to make it work for both mode?

**Comment by ghiscoding** [comment]
@xueran 
Great PR, I wasn't sure on how to name this option but I guess your option name is fine. However, we do need documentation for this though, so if you could add docs in the version [readme](https://github.com/lerna-lite/lerna-lite/tree/main/packages/version#readme). The other last missing piece would be to update the Lerna-Lite [JSON Schema](https://github.com/lerna-lite/lerna-lite/blob/main/packages/cli/schemas/lerna-schema.json) but I could do it in a separate PR if it's too much trouble. Thanks

File: packages/core/src/utils/collect-updates/collect-updates.ts (hunk 2 above)
Comment (xueran) [reply packages/core/src/utils/collect-updates/collect-updates.ts:50]: yes, it is only work for the `independent` mode, I'm not sure whether it will be better to make it work for both modes

**Comment by xueran** [comment]
> @xueran Great PR, I wasn't sure on how to name this option but I guess your option name is fine. However, we do need documentation for this though, so if you could add docs in the version [readme](https://github.com/lerna-lite/lerna-lite/tree/main/packages/version#readme). The other last missing piece would be to update the Lerna-Lite [JSON Schema](https://github.com/lerna-lite/lerna-lite/blob/main/packages/cli/schemas/lerna-schema.json) but I could do it in a separate PR if it's too much trouble. Thanks

ok, please wait a moment

File: packages/core/src/utils/collect-updates/collect-updates.ts (hunk 2 above)
Comment (ghiscoding) [reply packages/core/src/utils/collect-updates/collect-updates.ts:50]: it's probably better to support it for both, I'm also wondering why the name `describe`? is this option only used for reading a tag list or will it also be used to add tag names? If it does add tag then I'm not sure of the word `describe`

Comment (xueran) [reply packages/core/src/utils/collect-updates/collect-updates.ts]: this option only used for reading a tag list, and now I have revised it to support for both.

**Comment by xueran** [comment]
The test fail reason is `Cannot find module 'yargs/yargs' from 'packages/listable/src/__tests__/listable-options.spec.ts'`.
But I can run successfully locally, and I didn't change this file.

**Comment by ghiscoding** [comment]
@xueran it seems that I forgot to add yargs to the dev dependencies section of that package, not sure why it was failing before but if you just add it then it should work as intended

https://github.com/lerna-lite/lerna-lite/blob/f2c91cee5a06236b8fc229eb1aac7591d91398ef/packages/listable/package.json#L35-L40

just add 

```json
"devDependencies": {
    "yargs": "^17.7.1"
}
```

File: packages/core/src/utils/collect-updates/collect-updates.ts (hunk 2 above)
Comment (ghiscoding) [reply packages/core/src/utils/collect-updates/collect-updates.ts]: ok great then `describe` should be fine

File: packages/core/src/utils/collect-updates/collect-updates.ts [hunk 3]
```diff
@@ -44,16 +45,14 @@ export function collectUpdates(
       : new Map(filteredPackages.map(({ name }) => [name, packageGraph.get(name)]));
 
   let committish = commandOptions.since;
-  const tagPattern = isIndependent ? '*@*' : '';
+  const tagPattern = describeTag ? describeTag : isIndependent ? '*@*' : '*v*';
```
Comment (ghiscoding) [inline packages/core/src/utils/collect-updates/collect-updates.ts:48]: what's the reason of the change to `*v*` for the version mode? It was empty before and I would rather not change the default behavior if possible, did you see if `*v*` was used anywhere else and that's why you added it?

Comment (xueran) [reply packages/core/src/utils/collect-updates/collect-updates.ts:48]: Because  want to support two modes, if it empty, then can only make it effective in independent mode, otherwise there will be bugs in non-independent mode. 

the default setting is `*v*`, because in non-independent mode, the default tag is `v *.*.*`

Comment (ghiscoding) [reply packages/core/src/utils/collect-updates/collect-updates.ts:48]: but where did you find that it`s the default for version mode? Even if `v` is the default tag, the user can change it via [`--tag-version-prefix`](https://github.com/lerna-lite/lerna-lite/tree/main/packages/version#--tag-version-prefix) and I don't want to break that, I really don't think `*v*` is really the default unless proven

Comment (xueran) [reply packages/core/src/utils/collect-updates/collect-updates.ts:48]: Sorry, I'm wrong.  if it empty, there won't be any bugs, but there won't be any left.

Comment (xueran) [reply packages/core/src/utils/collect-updates/collect-updates.ts:48]: Sorry, I'm wrong. There will be no bugs, just the same effect as the original code

Comment (ghiscoding) [reply packages/core/src/utils/collect-updates/collect-updates.ts:48]: Sorry I'm not sure to follow what you mean, I still think it should be empty string for version mode (when not using independent mode). Could you add the missing yargs devDeps so that we could at least see the unit tests passing, I wrote a comment earlier

Comment (xueran) [reply packages/core/src/utils/collect-updates/collect-updates.ts]: yargs devDeps  is added, and  empty string for version mode is ok now

**Comment by ghiscoding** [comment]
thanks for the change, I'll do some more checks before merging tomorrow after work and a possible release by the end of the week. Thanks for the contribution

**Comment by ghiscoding** [comment]
@xueran Thanks again for the contribution, the new option is now available in the new [v1.17.0](https://github.com/lerna-lite/lerna-lite/releases/tag/v1.17.0) release. 

Cheers ☕⭐🎉 🚀
